src/main/java/picard/sam/UmiMetrics.java

```diff
@@ -103,64 +87,27 @@ public UmiMetrics(final int length, final int observedUniqueUmis, final int infe
         INFERRED_UMI_ENTROPY = effectiveLengthOfInferredUmis;
         OBSERVED_UMI_ENTROPY = effectiveLengthOfObservedUmis;
         UMI_BASE_QUALITIES = estimatedBaseQualityOfUmis;
-        UMI_COLLISION_EST = expectedUmiCollisions;
-        UMI_COLLISION_Q = umiCollisionQ;
     }
 
-    public void calculateDerivedFields() {
+    public void calculateDerivedFields(final Histogram<String> observedUmis, Histogram<String> inferredUmis, long observedUmiBases) {
```

Comment:
These should be private members in the metric...no need to pass them into a function.